### ISSUE TITLE
Gtk4 applications seem to crash when trying to inhibit sleep/idle inhibit

### ISSUE THREAD

**Issue by GorrillaRibs** [swaywm/sway#6772]
- **Sway Version:**
Gtk applications seem to crash when trying to inhibit sleep/idle inhibit

- **Debug Log:**
https://gist.github.com/GorrillaRibs/0a046c72676e2343a9e9f172e743f703

- **Configuration File:**
Default config seems to reproduce this on arch

- **Description:**

 I used celluloid (gtk frontend for mpv) as an example, but any gtk4 application that uses idle_inhibit seems to do it (I tried 'clapper' as well, which crashed with the same error). This might just be a gtk bug, but figured I'd mention it here in case anyone else has run into it.

Gdk:ERROR:../gtk/gdk/wayland/gdksurface-wayland.c:2286:gdk_wayland_toplevel_inhibit_idle: assertion failed: (wayland_toplevel->idle_inhibitor && wayland_toplevel->idle_inhibitor_refcount > 0)
Bail out! Gdk:ERROR:../gtk/gdk/wayland/gdksurface-wayland.c:2286:gdk_wayland_toplevel_inhibit_idle: assertion failed: (wayland_toplevel->idle_inhibitor && wayland_toplevel->idle_inhibitor_refcount > 0)
[1]    241368 IOT instruction (core dumped)  celluloid Downloads/skelly.mp4

  - The steps you took in plain English to reproduce the problem.

**Comment by emersion** [comment]
Well this sounds like a GTK bug. Please report it to them.

**Comment by codicodi** [comment]
Seems like there's a fix pending for gtk

https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4385
